torch/_subclasses/fake_tensor.py

```diff
@@ -613,24 +629,39 @@ def check_non_fake_tensor(x):
                 except NotImplementedError as not_implemented_error:
                     if not self.allow_fallback_kernels:
                         raise not_implemented_error
-                    r = run_fallback_kernel(func, args, kwargs, not_implemented_error)
+                    return run_fallback_kernel(
+                        self, func, args, kwargs, not_implemented_error
+                    )
 
             # TODO: handle non-kwarg devices
             assert func not in _device_not_kwarg_ops, f"NYI: {func}"
 
+            # Lazily initialized, in case there are no tensor returns
+            common_device = None
```

Comment:
fwiw, I stole this part of your code to get https://github.com/pytorch/pytorch/pull/82449 to pass hehe (otherwise aten.maximum_value or something along those lines with no Tensor inputs would fail)